installers/AppVeyor.hs

```diff
@@ -0,0 +1,160 @@
+{-# LANGUAGE NoImplicitPrelude, OverloadedStrings, LambdaCase, RecordWildCards, DeriveDataTypeable, DeriveGeneric #-}
+
+module AppVeyor (downloadCardanoSL, AppVeyorError(..)) where
+
+import Universum hiding (get)
+import Data.Aeson
+import GitHub hiding (URL)
+import GitHub.Endpoints.Repos.Statuses
+import qualified GitHub as GH
+import qualified Data.ByteString.Lazy.Char8 as L8
+import qualified Data.Text as T
+import Network.URI
+import Data.Maybe (mapMaybe)
+import Network.Wreq
+import Data.Aeson.Lens
+import Lens.Micro
+import Turtle.Format (printf, (%), d, s, w, Format, makeFormat)
+import GHC.Generics
+
+-- | Gets CardanoSL.zip corresponding to the src json revision from AppVeyor CI
+downloadCardanoSL :: FilePath -> IO L8.ByteString
+downloadCardanoSL srcJson = do
+  src@CardanoSource{..} <- getCardanoRev srcJson
```

Comment:
I think this begs to be factored into a function of type `SrcSpec -> IO L8.ByteString`, what do you think?

Reply:
Why?

Reply:
To have more orthogonal, reusable code, I guess.